synapse/lib/httpapi.py

```diff
@@ -442,7 +486,8 @@ def on_connection_close(self):
         if hasattr(self, 'task'):
             self.task.cancel()
 
-    async def _reqValidOpts(self, opts):
+    async def _reqValidOpts(self, opts) -> tuple[bool, dict]:
```

Comment:
this type declaration is wrong

Reply:
That was an artifact when I thought I could handle the pre-requisites in one shot on the Cortex; but thats not readily achievable without always incurring an additional round trip to the cortex with each call; so I abandoned that approach.